### PR TITLE
feat(discover): Add option to filter for project

### PR DESCRIPTION
**Pull request by wmak**
- This is to help improve usability when an organization have a lot of projects, if many of them have never sent a transaction we can just exclude those so the query we send to snuba is far smaller

### CODE REVIEW

File: src/sentry/search/events/builder/base.py [hunk 1]
```diff
@@ -570,7 +573,8 @@ def resolve_params(self) -> list[WhereType]:
         # complain on an empty list which results on no data being returned.
         # This change will prevent calling Snuba when no projects are selected.
         # Snuba will complain with UnqualifiedQueryError: validation failed for entity...
-        if not self.params.project_ids:
+        project_ids = self.resolve_projects()
+        if not project_ids:
```
Comment (Zylphrex) [inline src/sentry/search/events/builder/base.py:577]: Could this result in poor UX? For example, users go on to a page like discover before sending an event then they get this error saying they need to select an project when that problem is that they need to send an event first.

File: src/sentry/search/events/builder/discover.py [hunk 2]
```diff
@@ -86,6 +87,12 @@ def resolve_field(self, raw_field: str, alias: bool = False) -> Column:
 
         return super().resolve_field(raw_field, alias)
 
+    def resolve_projects(self) -> list[int]:
+        if options.get("sentry.search.events.project.check_event"):
+            return [proj.id for proj in self.params.projects if proj.flags.has_transactions]
```
Comment (Zylphrex) [inline src/sentry/search/events/builder/discover.py:92]: Can't discover also query for errors?

File: src/sentry/search/events/builder/base.py (hunk 1 above)
Comment (wmak) [reply src/sentry/search/events/builder/base.py:577]: 🤔 I think I'll just add `You need to select at least one project with data`

File: src/sentry/search/events/builder/discover.py (hunk 2 above)
Comment (wmak) [reply src/sentry/search/events/builder/discover.py]: Good callout, changed this to treat the condition differently based on the dataset selected
Discover -> either an error or a txn has been sent
Events -> an error has been sent
Txn/IssuePlatform -> a txn has been sent

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**